### PR TITLE
BUG: Doctest doesn't have a SHOW_WARNINGS directive.

### PR DESCRIPTION
**Pull request by charris**

### CODE REVIEW

File: numpy/core/_ufunc_config.py [hunk 1]
```diff
@@ -154,7 +154,7 @@ def geterr():
     --------
     >>> np.geterr()
     {'divide': 'warn', 'over': 'warn', 'under': 'ignore', 'invalid': 'warn'}
-    >>> np.arange(3.) / np.arange(3.)  # doctest: +SHOW_WARNINGS
```
Comment (bsipocz) [inline numpy/core/_ufunc_config.py:157]: ahh, yes, this was a premature change that requires pytest-doctestplus (@seberg was originally working on the #23812 branch)

File: numpy/core/_ufunc_config.py [hunk 2]
```diff
@@ -154,7 +154,7 @@ def geterr():
...
+    >>> np.arange(3.) / np.arange(3.)
     array([nan,  1.,  1.])
     RuntimeWarning: invalid value encountered in divide
```
Comment (bsipocz) [inline numpy/core/_ufunc_config.py:159]: I believe the exception needs to be removed here in here.

```suggestion
```

File: numpy/core/_ufunc_config.py (hunk 1 above)
Comment (charris) [reply numpy/core/_ufunc_config.py:157]: @bsipocz Hmm, removing that fails differently on the refguide checks. I think the best short term fix is to skip that test. Could you make sure to add it back in your PR?

File: numpy/core/_ufunc_config.py (hunk 2 above)
Comment (seberg) [reply numpy/core/_ufunc_config.py:159]: I think the warning is correct, but maybe warnings are set up differently in the two runs.  With pytest, we have the problem that warnings are set to errors (when they should probably not be for the doctest part), and maybe the current checks don't handle them at all?

In any case, I tend towards the +SKIP.  If we forget about it... this function should maybe be hidden away from public anyway...

**Comment by seberg** [comment]
Not sure that SKIP is ideal, but I do think the warning should be in the docs (it would show the warning in a normal setup there), so SKIP is hopefully OK for now and unblocks CI.

Thanks for fixing this up Chuck!

**Pull request merged**